Library/Services/INfieldSurveySampleService.cs

```diff
@@ -67,6 +67,16 @@ public interface INfieldSurveySampleService
         /// <returns>number of deleted sample records</returns>
         Task<int> DeleteAsync(string surveyId, string respondentKey);
 
+        /// <summary>
+        /// delete a sample record
+        /// </summary>
+        /// <param name="surveyId">The id of the survey</param>
+        /// <param name="filters">Filters to be used in the delete operation</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="ArgumentException"></exception>
+        /// <returns>number of deleted sample records</returns>
+        Task<int> DeleteWithFiltersAsync(string surveyId, List<SampleFilter> filters);
```

Comment:
Across the solution, this kind of methods are named with "BY"
```suggestion
        Task<int> DeleteByFilterAsync(string surveyId, List<SampleFilter> filters);
```
the same for:
BlockWithFiltersAsync ->BlockByFilterAsync 
ResetWithFiltersAsync -> ResetByFilterAsync 
ClearByFilter**s**Async -> ClearByFilterAsync (Singular)